Origin: 92e86ba5787c42df8a057bdeac4560ca127535a4
From: Václav Doležal <vdolezal@redhat.com>
Date: Fri, 21 Feb 2020 14:02:51 +0100
Bug: https://lists.gnu.org/archive/html/screen-devel/2020-02/msg00011.html
Description: Expand d_xtermosc array in struct display

Commit c5db181 expands index range of "typ2" by one without expanding
affected arrays. d_xtermosc in struct display is one of these.

Related: c5db181b6e017cfccb8d7842ce140e59294d9f62
  (ansi: add support for xterm OSC 11)
Related: 68386dfb1fa33471372a8cd2e74686758a2f527b
  (Fix out of bounds access when setting w_xtermosc after OSC 49)

Signed-off-by: Václav Doležal <vdolezal@redhat.com>
---
 src/display.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/display.h
+++ b/display.h
@@ -112,7 +112,7 @@
   int	d_mousetrack;		/* set when user wants to use mouse even when the window
 				   does not */
 #ifdef RXVT_OSC
-  int   d_xtermosc[4];		/* osc used */
+  int   d_xtermosc[5];		/* osc used */
 #endif
   struct mchar d_lpchar;	/* missing char */
   struct timeval d_status_time;	/* time of status display */
